private/bufpkg/bufapimodule/module_reader.go

```diff
@@ -28,14 +28,28 @@ import (
 
 type moduleReader struct {
 	downloadClientFactory DownloadServiceClientFactory
+	tamperProofingEnabled bool
+}
+
+// WithTamperProofing configures whether the module reader supports tamper proofing.
+// If false, when downloading modules the manifest and blobs will be ignored.
+func WithTamperProofing(enabled bool) ModuleReaderOption {
```

Comment:
Generally these options have not accepted a boolean, and instead would be ie `WithTamperProofing()`